### ISSUE TITLE
[Menu] Menu items with `disabled` are not announced as `dimmed` by screen reader

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3168]
## Current behavior

Menu items that are disabled use `disabled` attribute which is not announced by screen reader. 
Right now our stories use additional information in visible text of menu item that says "disabled" or "(disabled)"

<img width="209" height="39" alt="Image" src="https://github.com/user-attachments/assets/a2224b68-4f3e-4402-bbc8-9c5307a3120b" />

<img width="162" height="43" alt="Image" src="https://github.com/user-attachments/assets/803f05d1-4281-4b9b-9ad7-a8928f55ff6a" />

This is helpful but may be insufficient if developer try to add disabled menu element without this visible information. This also creates issues in stories like [Tool bar](https://jolly-stone-0f7fffc03-3139.westeurope.6.azurestaticapps.net/?path=/story/actions-tool-bar--overflow) where "Copy" has correctly added `aria-disabled` when it's inside tool bar but it changes to `disabled` when it's inside menu. Then user loses information about "Copy" being dimmed/unavailable.  

<img width="403" height="304" alt="Image" src="https://github.com/user-attachments/assets/f14a11ea-9f5b-477b-96e7-6d67ff8d14f1" />

User can't reach `disabled` menu items with normal arrow shortcuts. But when they get information about how many items there are inside menu and the number is not equal to items that can be reached with the most common way some users may investigate on other buttons and then it's good to inform on each unavailable menu item that it is indeed disabled.

## Suggested solution

It's recommended to use `aria-disabled` on disabled menu items, so they would be correctly announced by screen reader when reached. 
It's then recommended to remove `disabled` attribute and replace its function with JS, CSS and `"tabindex='-1'"`.

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
